onnxruntime/core/providers/cpu/nn/conv_transpose.cc

```diff
@@ -53,30 +53,31 @@ inline void ComputeTransposePadAndOutputShape(
     }
     return;
   }
-    if (pad_type != AutoPadType::NOTSET) {
-      switch (pad_type) {
-          // We handle cases of AutoPadType::VALID and AutoPadType::SAME_UPPER/LOWER,
-          // the same way
-        case AutoPadType::VALID:
-        case AutoPadType::SAME_UPPER:
-        case AutoPadType::SAME_LOWER:
-          *pad_head = 0;
-          *pad_tail = 0;
-          *out_size = (in_size - 1) * stride + kernel + dilation - 1 + adj;
-          break;
-        default:
-          throw NotImplementedException("pad type not supported");
-      }
-    } else {
-      *out_size =
-          (in_size - 1) * stride + kernel + dilation - 1 + adj - *pad_head - *pad_tail;
+  if (pad_type != AutoPadType::NOTSET) {
+    switch (pad_type) {
+        // We handle cases of AutoPadType::VALID and AutoPadType::SAME_UPPER/LOWER,
+        // the same way
+      case AutoPadType::VALID:
+      case AutoPadType::SAME_UPPER:
+      case AutoPadType::SAME_LOWER:
+        *pad_head = 0;
+        *pad_tail = 0;
+        *out_size = (in_size - 1) * stride + kernel + dilation - 1 + adj;
+        break;
+      default:
+        throw NotImplementedException("pad type not supported");
     }
+  } else {
+    *out_size =
+        (in_size - 1) * stride + kernel + dilation - 1 + adj - *pad_head - *pad_tail;
+  }
 }
 
-Status ConvTransposeBase::PrepareForCompute(OpKernelContext* context, bool has_bias, ConvTransposeBase::Prepare& p) const {
-  const auto* X = context->Input<Tensor>(0);
-  const auto* F = context->Input<Tensor>(1);
-  const Tensor* B = has_bias ? context->Input<Tensor>(2) : nullptr;
+Status ConvTransposeBase::PrepareForCompute(OpKernelContext* context, bool has_bias, ConvTransposeBase::Prepare& p, bool dynamic_padding) const {
+  const Tensor* X = context->Input<Tensor>(0);
+  const Tensor* F = context->Input<Tensor>(1);
+  const Tensor* Pads = dynamic_padding ? context->Input<Tensor>(2) : nullptr;
```

Comment:
In the schema, Pads is optional. But it seems that the kernel assume pads exist.